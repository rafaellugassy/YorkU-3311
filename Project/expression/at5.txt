exists
start_of_set_enumeration 
conjunction
bool_value (true)
bool_value (false)
disjunction
bool_value (false)
bool_value (false)
exclusive_or -- true xor false is true
bool_value (true)
bool_value (false)
end_of_set_enumeration
analyze
simplify