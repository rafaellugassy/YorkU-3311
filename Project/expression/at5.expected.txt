  Expression currently specified: ?
  Report: Expression is initialized.
->exists
  Expression currently specified: (|| ?)
  Report: OK.
->start_of_set_enumeration
  Expression currently specified: (|| {?})
  Report: OK.
->conjunction
  Expression currently specified: (|| {(? && nil)})
  Report: OK.
->bool_value(True)
  Expression currently specified: (|| {(True && ?)})
  Report: OK.
->bool_value(False)
  Expression currently specified: (|| {(True && False), ?})
  Report: OK.
->disjunction
  Expression currently specified: (|| {(True && False), (? || nil)})
  Report: OK.
->bool_value(False)
  Expression currently specified: (|| {(True && False), (False || ?)})
  Report: OK.
->bool_value(False)
  Expression currently specified: (|| {(True && False), (False || False), ?})
  Report: OK.
->exclusive_or
  Expression currently specified: (|| {(True && False), (False || False), (? xor nil)})
  Report: OK.
->bool_value(True)
  Expression currently specified: (|| {(True && False), (False || False), (True xor ?)})
  Report: OK.
->bool_value(False)
  Expression currently specified: (|| {(True && False), (False || False), (True xor False), ?})
  Report: OK.
->end_of_set_enumeration
  Expression currently specified: (|| {(True && False), (False || False), (True xor False)})
  Report: OK.
->analyze
  Expression currently specified: (|| {(True && False), (False || False), (True xor False)})
  Report: (|| {(True && False), (False || False), (True xor False)}) is type-correct.
->simplify
  Expression currently specified: (|| {(True && False), (False || False), (True xor False)})
  Report: True
